src/perfui/tmp/perf_config.txt
{'client.response.endlist.tag': '#EXT-X-ENDLIST', 'test.client.request.timeout': '7', 'test.case.survival': '54000', 'db.vex.result.store.enable': 'True', 'psn.fake.send': 'False', 'client.response.ad.mid.roll.ts.number': '135', 'test.index.asset.content.size': '10k', 'client.response.ad.pre.roll.ts.number': '45', 'test.result.report.error.file': 'load-test-error.txt', 'golden_config_file': '/Users/xieyanyang/work/learning/PerfTest/src/perf/test/vod/config-golden.properties', 'test.machine.pubkey': '/Users/xieyanyang/work/ttbj/ttbj-keypair.pem', 'psn.send': 'False', 'db.database': 'perftest', 'test.use.sap': 'True', 'db.vex.result.table': 'perf_result', 'client.response.check.percent': '0.5', 'db.host': 'localhost', 'client.response.check.when.running': 'True', 'test.case.counter.dump.interval': '60', 'client.response.media.sequence': '18', 'test.case.name': 'vod-perf', 'bitrate.response.counter': '0,200,500,1000,2000,3000,6000,12000', 'psn.fake.position': '1,76,151,226,301', 'test.bitrate.serial.time.gap': '300', 'client.response.ad.post.roll.ts.number': '45', 'index.response.counter': '0,200,500,1000,2000,3000,6000,12000', 'test.machine.username': 'root', 'test.result.log.file': 'load-test.log', 'test.result.collect.before.now': '4', 'db.user': 'root', 'test.case.server.instance.number': '2', 'test.require.sap': 'True', 'test.result.report.traced.dir': 'load-test-trace', 'test.client.request.retry.count': '3', 'test.execute.process.number': '1', 'task.apschdule.queue.misfire.time': '300', 'log.export.thirdparty': 'False', 'task.apschdule.threadpool.core.threads': '150', 'psn.apschdule.threadpool.max.threads': '100', 'client.response.media.tag': '#EXT-X-MEDIA-SEQUENCE', 'test.result.report.dir': '/tmp/load-test-result', 'db.vex.config.table': 'vex_config', 'psn.endall.send': 'False', 'test.use.iframe': 'True', 'test.case.type': 'VOD_T6', 'test.client.vip.latest.segment.range': '100~119', 'test.machine.hosts': '54.169.51.77', 'psn.fake.tracking.id': '1-2-3-4', 'log.level': 'INFO', 'test.bitrate.asset.content.size.merged': '300k', 'db.vex.config.enable': 'False', 'test.use.fake.manifest': 'True', 'task.apschdule.queue.max': '100000', 'test.case.concurrent.number': '4', 'test.result.report.delta.file': 'delta-report.txt', 'test.bitrate.serial': 'False', 'test.result.report.file': 'load-test-report.txt', 'test.result.report.error.dir': 'load-test-error', 'db.password': '111111', 'client.response.ad.tag': 'espn', 'test.bitrate.asset.content.size': '150k', 'project.name': 'VEX-Core', 'psn.message.sender.position': '0,0.1,0.2,0.3,0.4,0.5,0.6,0.7,0.8,0.9,1', 'test.bitrate.request.number': '2', 'client.response.ad.mid.roll.position': '225,450,675', 'client.response.asset.tag': 'vod', 'psn.endall.position': '2000', 'psn.apschdule.threadpool.core.threads': '100', 'psn.apschdule.queue.max': '100000', 'client.response.content.segment.number': '900', 'test.client.location.number': '500', 'test.machine.port': '22', 'client.response.content.segment.time': '2', 'client.response.error.dump': 'False', 'client.response.drm.tag': '4g8W0yS1bDxiM8ejG4', 'test.case.warmup.period.minute': '10', 'test.case.content.names': 'vod_test_[1~15000]', 'test.result.report.traced.file': 'load-test-response.txt', 'test.bitrate.request.random': 'True', 'psn.receiver.host': '54.169.51.181', 'task.apschdule.threadpool.max.threads': '150', 'db.port': '3306', 'psn.tag': '#EXT-X-CUE', 'psn.apschdule.queue.misfire.time': '36000', 'test.result.report.delta.dir': 'delta-reports', 'test.client.request.retry.delay': '1', 'test.client.zone.number': '500', 'project.version': '2.8.0-SNAPSHOT', 'psn.receiver.port': '80'}

Index response summary  
  Test Duration (second): 900 
  Request concurrent    : 23  
  Request In Total      : 21570
  Request Succeed       : 21570
  Request Failure       : 0
  Request Succeed Rate  : 100.00%
  Response Average Time : 14
  Response Failure      : 0
  Response Time Distribution
      0-200      millisecond: 21562
      200-500    millisecond: 5
      500-1000   millisecond: 2
      1000-2000  millisecond: 1
      2000-3000  millisecond: 0
      3000-6000  millisecond: 0
      6000-12000 millisecond: 0

Bitrate response summary
  Test Duration (second): 900 
  Request concurrent    : 47  
  Request In Total      : 43121
  Request Succeed       : 43121
  Request Failure       : 0
  Request Succeed Rate  : 100.00%
  Response Average Time : 25
  Response Failure      : 2163
  Response Time Distribution
      0-200      millisecond: 43116
      200-500    millisecond: 4
      500-1000   millisecond: 1
      1000-2000  millisecond: 0
      2000-3000  millisecond: 0
      3000-6000  millisecond: 0
      6000-12000 millisecond: 0
      
69.209.207.113   : Manifest ad pre-roll number is 0, not the same as expected number 45
34.110.108.104   : Manifest ad pre-roll number is 0, not the same as expected number 45
36.223.229.104   : Manifest ad pre-roll number is 0, not the same as expected number 45
62.217.146.113   : Manifest ad pre-roll number is 0, not the same as expected number 45
43.42.206.100    : Manifest ad pre-roll number is 0, not the same as expected number 45

      

src/PerfTestUI/settings.py
"""
Django settings for PerfTestUI project.

Generated by 'django-admin startproject' using Django 1.8.

For more information on this file, see
https://docs.djangoproject.com/en/1.8/topics/settings/

For the full list of settings and their values, see
https://docs.djangoproject.com/en/1.8/ref/settings/
"""

# Build paths inside the project like this: os.path.join(BASE_DIR, ...)
import os

BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))


# Quick-start development settings - unsuitable for production
# See https://docs.djangoproject.com/en/1.8/howto/deployment/checklist/

# SECURITY WARNING: keep the secret key used in production secret!
SECRET_KEY = 'vjwlg(t+h+ed#i&oa^4mc!03e^wmj11q$xpp1e9)v30ubu5*!)'

# SECURITY WARNING: don't run with debug turned on in production!
DEBUG = True if __file__.find('/Users/xieyanyang') >= 0 else False
TEMPLATE_DEBUG = False

ALLOWED_HOSTS = [ '*', '127.0.0.1']


# Application definition

INSTALLED_APPS = (
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',
    'perfui',
)

MIDDLEWARE_CLASSES = (
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.common.CommonMiddleware',
    #'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django.contrib.auth.middleware.SessionAuthenticationMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    'django.middleware.clickjacking.XFrameOptionsMiddleware',
    'django.middleware.security.SecurityMiddleware',
)

ROOT_URLCONF = 'PerfTestUI.urls'

TEMPLATES = [
    {
        'BACKEND': 'django.template.backends.django.DjangoTemplates',
        'DIRS': [os.path.join(BASE_DIR, "templates/")],
        'APP_DIRS': True,
        'OPTIONS': {
            'context_processors': [
                'django.template.context_processors.debug',
                'django.template.context_processors.request',
                'django.contrib.auth.context_processors.auth',
                'django.contrib.messages.context_processors.messages',
            ],
        },
    },
]

WSGI_APPLICATION = 'PerfTestUI.wsgi.application'


# Database
# https://docs.djangoproject.com/en/1.8/ref/settings/#databases
DATABASES = {
     'default': {
          'ENGINE': 'django.db.backends.mysql',  # Add 'postgresql_psycopg2', 'postgresql', 'mysql', 'sqlite3' or 'oracle'.
          'NAME': 'perftest',  # Or path to database file if using sqlite3.
          'USER': 'root',  # Not used with sqlite3.
          'PASSWORD': '111111',  # Not used with sqlite3.
          'HOST': '127.0.0.1',  # Set to empty string for localhost. Not used with sqlite3.
          # 'HOST': '54.169.51.190',
          'PORT': '3306',  # Set to empty string for default. Not used with sqlite3.
     }
}


# Internationalization
# https://docs.djangoproject.com/en/1.8/topics/i18n/
LANGUAGE_CODE = 'en-us'

TIME_ZONE = 'UTC'

USE_I18N = True

USE_L10N = True

USE_TZ = True


# Static files (CSS, JavaScript, Images)
# https://docs.djangoproject.com/en/1.8/howto/static-files/

STATIC_URL = '/static/'
STATIC_ROOT = BASE_DIR + '/static'

LOG_FILE = BASE_DIR + "/perfui.log"

LOGGING = {
    'version': 1,
    'disable_existing_loggers': False,
    'formatters': {
        'standard': {
            'format' : "[%(asctime)s] %(levelname)s [%(name)s:%(lineno)s] %(message)s",
            'datefmt' : "%d/%b/%Y %H:%M:%S"
        },
    },
    'handlers': {
        'mail_admins': {
            'level': 'ERROR',
            'class': 'django.utils.log.AdminEmailHandler'
        },
       'logfile': {
            'level':'INFO',
            'class':'logging.handlers.RotatingFileHandler',
            'filename': LOG_FILE,
            'maxBytes': 1024 * 1024 * 5,
            'backupCount': 5,
            'formatter': 'standard',
        },
        'console':{
            'level':'DEBUG',
            'class':'logging.StreamHandler',
            'formatter': 'standard'
        },
    },
    'loggers': {
        'django.request': {
            'handlers': ['mail_admins'],
            'level': 'ERROR',
            'propagate': True,
        },
        '': {
            'handlers': ['logfile', 'console'],
            'level': 'DEBUG',
            'propagate': True,
        },
        'django.db': {
            # # django also has database level logging
            'handlers': ['console'],
            
            # if level is debug, then sql will be showed into console
            'level': 'INFO',
            'propagate': False,
        },
    }
}

CACHES = {
    'default': {
        'BACKEND': 'django.core.cache.backends.locmem.LocMemCache',
        'LOCATION': 'unique-snowflake1',
        'options': {
            'MAX_ENTRIES': 1024,
        }
    },
    'memcache': {
        'BACKEND': 'django.core.cache.backends.memcached.MemcachedCache',
        # 'LOCATION': 'unix:/home/billvsme/memcached.sock',
        'LOCATION': '127.0.0.1:11211',
        'options': {
            'MAX_ENTRIES': 1024,
        }
    },
}